File: (03-28-2024)Container With Most Water/joshlin/solution.ts
function maxArea(height: number[]): number {
    let [l, r] = [0 , height.length - 1];
    let res = 0;
    while (l < r) {
        res = Math.max(res, Math.min(height[l], height[r]) * (r - l));
        if (height[l] < height[r]) l++; else r--;
    }
    return res;
};

File: (03-28-2024)Container With Most Water/joshlin/solution.c
int maxArea(int* height, int heightSize) {
    int l = 0, r = heightSize - 1, res = 0;
    while (l < r) {
        int area = (r - l) * (height[l] <= height[r] ? height[l] : height[r]);
        res = area >= res ? area : res;
        if (height[l] < height[r]) l++; else r--;
    }
    return res;
}